### PR TITLE
Add E2E teamplate as example

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I implemented automatic injection of the E2E Template with default settings into `examples/e2e` to achieve always up-to-date examples for our users and have and ability to reference those codes directly from the docs.

For clarity: my changes are only touching the CI part and this [automatic commit](https://github.com/zenml-io/zenml/pull/1766/commits/42ab82a7001ebfc16849c5c9f2bad7be4d11649a) brought in the whole example. If there are no changes to the checked-out template versus what we have already - no commit will be created.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by avishniakov** [comment]
@htahir1 can you add actions-user to CLA exclusions?

**Comment by avishniakov** [comment]
Hey, hey, can I get your reviews, please? There is no actual code change, just one CI added to automatically populate the example from the template.

File: .github/workflows/update-e2e-template.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          - '3.9'
+          - '3.10'
+          - '3.11'
+        required: false
+        default: '3.8'
+
+jobs:
+  update-e2e-template:
+    name: update-e2e-template
+    runs-on: ${{ inputs.os }}
```
Comment (strickvl) [inline .github/workflows/update-e2e-template.yml:40]: Will this run on multiple runners? If we're just updating + making a pull request, could it just run on a single runner?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -51,6 +51,14 @@ jobs:
...
       os: ${{ matrix.os }}
     secrets: inherit
+  
+  update-e2e-template:
+    needs: ubuntu-setup-python-environment
+    uses: ./.github/workflows/update-e2e-template.yml
+    with:
+      python-version: "3.8"
+      os: ubuntu-latest
+    secrets: inherit
```
Comment (avishniakov) [inline .github/workflows/ci.yml:61]: @strickvl , this is the answer :) It runs on 3.8 Ubuntu only - no matrix here

File: .github/workflows/update-e2e-template.yml (hunk 1 above)
Comment (avishniakov) [reply .github/workflows/update-e2e-template.yml:40]: Pointed to code below

**Review by strickvl — APPROVED** [review]
🙏🏻

**Comment by avishniakov** [comment]
@AlexejPenner @bcdurak are we good to merge or not yet?

**Review by bcdurak — APPROVED** [review]

**Review by AlexejPenner — APPROVED** [review]
:seal:

**Pull request merged**